### DISCUSSION TITLE
Not able to focus an element on Stripe

### DISCUSSION THREAD

**Discussion by Pablolo** [tailwindlabs/tailwindcss#2602] [Help] [answered]
Hello everyone!

On my React app, I have this piece of code inside a form:

`<div className="focus:border-orange-400 bg-white border appearance-none border-gray-400 rounded my-0 mx-auto w-full py-2 px-3 text-gray-700 leading-tight">
<CardElement />
                </div>`

Where <CardElement /> is part of [React Stripe Js](https://stripe.com/docs/stripe-js/react). Everything works as expected but not the "focus:border-orange-400".

As I learned from Stripe documentation, you can also do this:

`<CardElement
                  options={{
                    style: {
                      base: {
                        fontSize: '16px',
                        color: '#424770',
                        '::placeholder': {
                          color: '#aab7c4',
                        },
                      },
                    },
                  }}
                />`

But that way, I'm not able to use Tailwind. 

Any suggestions? 

Thank you very much,

**Comment by simonswiss** [comment]
Hi @Pablolo 👋

Since you're triggering focusing on the card element itself (and not the wrapping div), you might need to use the [focus-within pseudo-class variant (`focus-within:`)](https://tailwindcss.com/docs/pseudo-class-variants#focus-within) instead of `focus:`.

When the `CardElement` enters focus, the focus state is _within_ the wrapping div you want to alter styles for, so that's the pseudo-class variant you're likely looking for.

You will need to enable the `focus-within`variant for the `borderColor` key:

```javascript
// tailwind.config.js
module.exports = {
  // ...
  variants: {
    borderColor: ({ after }) => after(['focus-within']),
  },
}
```

Note that this feature is not supported in IE.

Hopefully that will help! If not, could you please provide a little code repo with the example so we can have a look?

**Comment by Pablolo** [comment, marked as answer]
Thank you very much @simonswiss!

It still didn't work like that, so in the end, I had to create an external CSS file and add these kinds of classes: https://stripe.com/docs/js/element/the_element_container

Like for example: 

`.StripeElement--focus {
    border-color: #80BDFF;
    outline:0;
    box-shadow: 0 0 0 .2rem rgba(0,123,255,.25);
    transition: border-color .15s ease-in-out, box-shadow .15s ease-in-out;
}`

**Comment by sbkl** [comment]
Hi,
In case someone looking, you can now use the classes property as below. Prefixes won't work:
```
<CardElement
  options={{
    classes: {
      base: "block w-full px-3 py-2 transition duration-300 mt-1 border border-gray-300 rounded-md shadow-sm text-sm",
      focus:
        "ring-indigo-500 ring-1 outline-none border-indigo-600",
      invalid: "text-red-500 border-red-600 ring-red-500",
    },
  }}
/>
```
